source/reactive-state.md

```diff
@@ -0,0 +1,153 @@
+
+You can use a global state to make components reactive using [Apollo Client's useReactiveVar hook](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/#storing-local-state-in-reactive-variables):
+
+```jsx
+import React from 'react';
+import { exampleOpenState } from './exampleOpenState.js'
+import { useReactiveVar } from '@apollo/client';
+[ . . . ]
+
+const ExampelModal = ({title, children}) => {
+
+  const handleOpen = () => {
+    exampleOpenState(true);
+  };
+  
+  const handleClose = () => {
+    exampleOpenState(false);
+  };
+  
+  return (
+    <div>
+      <Dialog onClose={handleClose} open={useReactiveVar(exampleOpenState.reactiveVar)}>
+          
+        <DialogTitle>{title}</DialogTitle>
+        
+        <DialogContent>{children}</DialogContent>
+        
+        <DialogActions>
+          <Button onClick={handleClose}>Close</Button>
+        </DialogActions>
+      
+      </Dialog>
+      
+      <Button onClick={handleOpen}>Open</Button>
+    </div>
+  );
+};
+```
+
+Now you may be thinking, "why can't I just use component state or `useState`?" Well, you can, but you need to consider two things:
+
+ 1. If the component is ever unmounted and mounted again, it will lose its state.
+ 2. If you update the state of a component, all of its child components will be remounted and will lose their state.
+
+Using the above example, if you open the modal, navigate to another route and then come back, the modal will no longer be open.
+
+Where you store your state has a lot to do with how persistent you need it to be:
+
+ * If you want that modal to start closed every time its mounted, store it in component state
+ * If you want the modal's open state to persist for the whole browser session, store it in reactive state
+ * If you want the state to persist even when the user comes back days later, store it in a field of the User document
+
+
+## Reactive state with an object value and optional schema
+
+In addition to simple scalars, you can also store objects in reactive state. If you also specify a schema, an error will be thrown if you try to set the state to a value that does not respect the schema.
+
+```js
+import { createReactiveState } from 'meteor/vulcan:core';
+
+const topBarSchema = {
+  messageId: {
+    type: String,
+    defaultValue: 'global.welcome',
+  },
+  showClose: {
+    type: Boolean,
+    defaultValue: true,
+  },
+  isOpen: {
+    type: Boolean,
+    defaultValue: false,
+  },
+  classes: {
+    type: Array,
+    optional: true,
+    arrayItem: {
+      type: String,
+    },
+  },
+};
+
+const topBarState = createReactiveState({ stateKey: 'topBarState', schema: topBarSchema });
+```
+
+We recommend that you export the returned state and import it where you need to use it. Alternatively, you can retrieve a registered state using `getReactiveState('topBarState')`. To get its value, use `topBarState()`, which in this case would be:
```

Comment:
What's the main argument for `createReactiveState` agains raw `makeVar` then?
It is definitely more "Vulcanish" because you can query state by name and have a global registry, but it could be nice to have another selling point.
The ability to reset globally is a first point, though I find it weird that Apollo doesn't allow it in the first place?
I guess the key advantage is also the use of a schema to validate the state shape?

Reply:
You are right, reactive state is a wrapper around `makeVar`, even more so with this latest rewrite. I think there are three selling points (and I will update the docs with this):

 * Above all Vulcan is a patterns library that provides a tested, simplified, and standardized way of implementing features like permissions and forms and interfacing with tools like Apollo and React
 * Reactive state adds schema validation to Apollo reactive vars
 * It also adds state mutation like React `setState()` - updating Apollo reactive vars replaces the whole value
 * It also adds the ability to reset all reactive states globally and does that automatically when the user logs out

Once again, I appreciate your feedback - it's helping me think things through and gives me ideas to improve the code and the docs.

Reply:
Awesome improvement thank you :) I should probably schedule a new minor/patch release soon